src/Libraries/PythonNodeModelsWpf/ScriptEditorWindow.xaml.cs

```diff
@@ -461,18 +492,24 @@ private void OnEngineChanged(object sender, System.Windows.Controls.SelectionCha
 
         private void OnScriptEditorWindowClosed(object sender, EventArgs e)
         {
-            completionProvider?.Dispose();
-            NodeModel.CodeMigrated -= OnNodeModelCodeMigrated;
-            this.Closed -= OnScriptEditorWindowClosed;
-            PythonEngineManager.Instance.AvailableEngines.CollectionChanged -= UpdateAvailableEngines;
+            var dybnamoView = Owner as DynamoView;
 
-            Analytics.TrackEvent(
-                Dynamo.Logging.Actions.Close,
-                Dynamo.Logging.Categories.PythonOperations);
-            
-            editText.TextChanged -= EditTextOnTextChanged;
-            FoldingManager.Uninstall(foldingManager);
-            foldingManager = null;
+            if (!dynamoViewModel.DockedWindows.ContainsKey(Uid))
```

Comment:
Would you add some comments?

Reply:
Added comment:
"When the script editor is docked, we don't to dispose the editor functionality.
Dispose it only when the window is closed."